salt/modules/boto3_route53.py

```diff
@@ -1000,3 +1003,448 @@ def change_resource_record_sets(HostedZoneId=None, Name=None,
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    if not Name:
+        raise SaltInvocationError('Name of healthcheck to update missing.')
+
+    keys = [
+        'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+        'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+        'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks'
+    ]
+    params = locals()
+    result = None
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    CallerReference = CallerReference if CallerReference else str(uuid.uuid4())  # future lint: disable=blacklisted-function
+    HealthCheckConfig = {}
+
+    for key in [x for x in keys if params.get(x) is not None]:
+        HealthCheckConfig[key] = params.get(key)
+
+    args = {'CallerReference': CallerReference, 'HealthCheckConfig': HealthCheckConfig}
+
+    while tries:
+        try:
+            result = conn.create_health_check(**args).get('HealthCheck')
+            conn.change_tags_for_resource(
+                ResourceType='healthcheck',
+                ResourceId=result['Id'],
+                AddTags=[{ 'Key': 'Name', 'Value': Name }],
+                )
+            return {'result': result }
+        except ClientError as e:
+            if tries and e.response.get('Error', {}).get('Code') == 'Throttling':
+                log.debug('Throttled by AWS API.')
+                time.sleep(3)
+                tries -= 1
+                continue
+            log.error('Failed to create healthcheck %s: %s', Name, six.text_type(e))
+            raise e
+
+    return result
+
+def update_health_check(Id, changes, region=None, key=None, keyid=None, profile=None, tries=20):
+    '''
+    Updates a health check with with passed changes request.
+
+    Name
+        The name to assign to newly created health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.update_health_check Id ... \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    if not Id:
+        raise SaltInvocationError('Id of healthcheck to update missing.')
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    args = changes or {}
+    args.update({'HealthCheckId': Id})
+
+    while tries:
+        try:
+            return {'result': conn.update_health_check(**args)}
+        except ClientError as e:
+            if tries and e.response.get('Error', {}).get('Code') == 'Throttling':
+                log.debug('Throttled by AWS API.')
+                time.sleep(3)
```

Comment:
same question here - should we prefer 3 seconds or increasing backoff?

Reply:
This is taken from already existing boto/boto3 modules.

Reply:
See other comment here, I think just a comment suggesting an alternate approach, but for now this code is fine.